frontend/src/plugins/__tests__/preferences.spec.js

```diff
@@ -0,0 +1,448 @@
+import { getters, loadFromLocalStorage, mutations } from '@/plugins/store/preferences'
+
+let originalLocalStorage
+beforeEach(() => {
+  originalLocalStorage = window.localStorage
+  window.localStorage = (() => {
+    let store = {}
+
+    return {
+      getStore: () => store,
```

Comment:
minor:
It is a stupid api design from the w3c/javascript spec that the store is exposed.
But if you implement the clear like it is now,
the store returned by getStore is not cleared by the clear function.

I don't know if it is like this in the specification or if it is intended.

Reply:
getStore isn't in the ecmascript spec I think, I added this for debugging. It isn't needed anymore, I will remove it.